mutation/frozen_mutation.cc

```diff
@@ -133,6 +133,30 @@ frozen_mutation freeze(const mutation& m) {
     return frozen_mutation{ m };
 }
 
+frozen_mutation freeze_gently_in_thread(const mutation& m) {
+    mutation_partition_serializer part_ser(*m.schema(), m.partition());
+
+    bytes_ostream bytes;
+    ser::writer_of_mutation<bytes_ostream> wom(bytes);
+
+    std::move(wom).write_table_id(m.schema()->id())
+                  .write_schema_version(m.schema()->version())
+                  .write_key(m.key())
+                  .partition([&] (auto wr) mutable {
+                    part_ser.write_gently(std::move(wr)).get();
```

Comment:
Making `write_gently()` also work with a thread would save some allocations and context switches. Maybe not worth it, the coroutine variant is more generic.

Reply:
It actually makes sense. We can pass a `is_preemptible` flag to write_serialized set when called in a thread and share its code.

Reply:
With that I can get rid of `freeze_gently_in_thread` altogether, just by adding a `is_preemptible` flag to `freeze`